### PR TITLE
Add keybind to reload Rust-Analyzer workspace

### PR DESCRIPTION
**Pull request by teozkr**
Allows you to pick up new Cargo dependencies without a (slooow) full server restart.

~Should not be merged before https://github.com/emacs-lsp/lsp-mode/pull/2728 is.~

### CODE REVIEW

**Comment by lebensterben** [comment]
One minor issue, this key binding should only be available when the LSP backend is rust-analyzer, and should emit a warning if it's not. (Preferably it also display a message when reloading workspace)

**Comment by teozkr** [comment]
@lebensterben Are these diagnostics ok?

File: layers/+lang/rust/funcs.el [hunk 1]
```diff
@@ -47,14 +47,25 @@
...
           "ss" 'lsp-rust-switch-server))
     (spacemacs//lsp-layer-not-installed-message)))
 
 (defun spacemacs//rust-setup-lsp-dap ()
   "Setup DAP integration."
   (require 'dap-gdb-lldb))
 
+(defun spacemacs//lsp-rust-analyzer-reload-workspace ()
+  (interactive)
+  (if (->> (lsp-workspaces)
```
Comment (lebensterben) [inline layers/+lang/rust/funcs.el:60]: Looks like `->>` requires `dash`.
It's best to avoid it or declare it as dependency explicitly.

**Comment by lebensterben** [comment]
> @lebensterben Are these diagnostics ok?

Yes on a second thought I think we can do this:
- When it's rust-analyzer, RELOAD the workspace and emit "Reloading workspace"
- When it's rls, RESTART the workspace and emit "RLS doesn't support reloading workspace, fallback to restarting workspace" (Something like this. It's idiot-proof.)

File: layers/+lang/rust/funcs.el (hunk 1 above)
Comment (teozkr) [reply layers/+lang/rust/funcs.el:60]: Looks like `dash` comes from the `spacemacs-bootstrap` layer, which implies (to me at least) that it's always available. `ivy-spacemacs-help` doesn't seem to mention any other layer explicitly depending on `dash`, from what I can tell.

Comment (lebensterben) [reply layers/+lang/rust/funcs.el:60]: That should be okay then.

**Comment by teozkr** [comment]
RLS seems to pick up Cargo.toml changes immediately, without requiring any manual reload. So maybe the correct behaviour here would actually be to do nothing?

**Comment by lebensterben** [comment]
If so, then you are right. And no need to make it an error. A message that no need to reload manually for rls should suffice.

**Comment by lebensterben** [comment]
We probably can do something even better. We can add a layer variable which toggles auto-reloading of workspace after user called cargo-process-add.

**Comment by teozkr** [comment]
I agree that that would be nice, but hooking into `cargo-process` seems like it'd require someone more familiar with elisp than me.. :/

**Comment by lebensterben** [comment]
@teozkr No problem I will do that in a separate commit. I hope your PR to lsp-mode is merged ASAP.

**Comment by teozkr** [comment]
It was, a few hours ago!

**Review by smile13241324 — APPROVED** [review]
👍

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Comment by lebensterben** [comment]
It comes to me that, when I added `lsp-rust-switch-server` under `SPC m s s`, instead of `SPC m b s`, the motivation was if otherwise, `SPC m b s` binding persists when you go to other major mode with lsp backend.

I haven't found a decent way to remove `SPC m b s`.
This PR adds a new binding `SPC m b R`, it would have the same issue.

@smile13241324 What do you think?

**Pull request closed without merging**